spark/src/main/scala/org/apache/comet/CometExecIterator.scala

```diff
@@ -84,6 +87,30 @@ class CometExecIterator(
   private var currentBatch: ColumnarBatch = null
   private var closed: Boolean = false
 
+  private def getMemoryLimitPerTask(conf: SparkConf): Long = {
+    val numCores = numDriverOrExecutorCores(conf).toFloat
+    val maxMemory = CometSparkSessionExtensions.getCometMemoryOverhead(conf)
```

Comment:
We should be using the total executor memory here rather than just the overhead memory. Without this change I was unable to run any benchmarks.

```suggestion
    val maxMemory = ConfigHelpers
      .byteFromString(conf.get("spark.executor.memory", "1024MB"), ByteUnit.BYTE)
```

Reply:
I thought that overhead memory is the amount of native memory reserved for comet to use. The actual memory usage will be doubled if we reserve the same amount of executor memory for comet.

Reply:
That's a good point. The default overhead amount was way too low for the benchmarks to run. Using the executor amount is probably overkill. I'll give this some more thought.